quetz/main.py

```diff
@@ -908,27 +907,64 @@ def get_channel_members(
     auth.assert_list_channel_members(channel.name)
     member_list = dao.get_channel_members(channel.name)
 
-    return member_list
+    return [
+        rest_models.PostMember(
+            username=member.user.username,
+            role=member.role,
+        )
+        for member in member_list
+    ]
```

Comment:
I'm not really happy with having to change the return type here. Again, the main issue here is that `post_channel_members` and  `get_channel_members` disagree on whether the role can be `None`.

An alternative would be to allow `role=None` in the `rest_models.Member` class. It would also be possible to completely ban `role=None` for channel members, but since there is a difference between being a channel member with `role=None` and not being a channel member, I would advise against that.